### PR TITLE
Fixed for supporting flow

### PR DESCRIPTION
**Pull request by ifsnow**
It is the same as the [PR](https://github.com/facebook/react-native/pull/22212) that I submitted in react-native repository.

I hope this helps.

### CODE REVIEW

**Comment by cpojer** [comment]
Wow, nice. Thank you! I'll leave it to @Krizzu to review and merge this PR. :)

File: lib/AsyncStorage.js [hunk 1]
```diff
@@ -21,6 +20,21 @@ const RCTAsyncStorage =
   NativeModules.RNC_AsyncSQLiteDBStorage ||
   NativeModules.AsyncLocalStorage;
 
+type ReadOnlyArrayString = $ReadOnlyArray<string>;
```
Comment (Krizzu) [inline lib/AsyncStorage.js]: What about exporting this to different file, say `types/flow.js` and then include it in `.flowconfig`?

Comment (ifsnow) [reply lib/AsyncStorage.js]: @Krizzu 
There is some troubles when using `.flowconfig`. For example, hyperlink(Go to definition) in vscode does not work properly. 

I think the approach below looks ok.

- Added `AsyncStorageType.js` to define types and separated it from `AsyncStorage.js`.
- Added `AsyncStorageType.js.flow` to provide more clean autocomplete.

Let me know what you think about this.

Comment (Krizzu) [reply lib/AsyncStorage.js]: On the second thought, having types defined in the same file is better than have 2 different files with types. Thought extracting flow types into another file would be better supported by VScode.

If you would revert 1 commit back, I'd be more than happy to merge it 👍 
Sorry for confusion!

Comment (ifsnow) [reply lib/AsyncStorage.js]: @Krizzu Sure. I had a similar idea. I modified it as you said.

Comment (Krizzu) [reply lib/AsyncStorage.js]: Great, thank You.
In that case, where we have Types in the same file, do we still need `.js.flow` file? Any advantages to having it?
Also, can you rebase with `master`, please?

Comment (ifsnow) [reply lib/AsyncStorage.js]: @Krizzu With `AsyncStorage.js.flow`, we can use more organized auto-completion in VSCode, but I realized there was a problem with the function to jump directly to `AsyncStorage.js` to check the code.

It seemed to have more disadvantages than merits, so I decided to delete `AsyncStorage.js.flow`. Sorry for confusion!

I finished what you said.

**Pull request merged**